vortex-array/src/array/sparse/mod.rs

```diff
@@ -45,9 +44,15 @@ impl SparseArray {
         indices_offset: usize,
         fill_value: Scalar,
     ) -> VortexResult<Self> {
-        if !matches!(indices.dtype(), &DType::IDX) {
-            vortex_bail!("Cannot use {} as indices", indices.dtype());
-        }
+        let indices_dtype = indices.dtype();
+        let indices_ptype = if !matches!(indices_dtype, &DType::IDX) {
+            vortex_bail!("Cannot use {} as indices", indices_dtype);
+        } else {
```

Comment:
nit: I think old logic was more readable, can just do `let indices_ptype = PType::try_from(&DType::IDX)` which is also easier for the compiler to optimize

Reply:
DType::IDX is a constant so you can just inline the ptype here?